### DISCUSSION TITLE
Resolving #252 Discussion

### DISCUSSION THREAD

**Discussion by Luke-Py-Walker** [atlassian/github-for-jira#375] [General]
Hey,

Here are the changes that need to be made: 

1. Pass reviewer data from the `context` into the `pull-request` transform. [See](https://github.com/integrations/jira/blob/487d69eb86cc260ab834c5bb899701dcf7b8d062/lib/github/pull-request.js#L8) 
2. Parse the reviewer data and add it to the `pull-request` transform [here](https://github.com/integrations/jira/blob/master/lib/transforms/pull-request.js#L62). Format the reviewer data using the [jira documentation](https://developer.atlassian.com/cloud/jira/software/rest/api-group-development-information/#api-devinfo-0-10-bulk-post).

Hopefully, it's that simple! 

One open-ended question that I need to answer through experimentation: How many reviewers can I pass into the Jira `pullRequests.reviewers` list? I have a feeling it's only one, but 🤞 .

**Comment by Luke-Py-Walker** [comment]
I'm having trouble with my GitHub configuration in Jira. Its sync status has been stuck at `PENDING` for over 45 min. I've only selected 2 little repo's (with only a few commits) to sync. Any suggestions for troubleshooting? @zmoazeni @tebriel

**Reply by Luke-Py-Walker** [reply]
Uninstalled the github app and jira app hoping to have everything working with a clean slate... and now I'm getting `{ error: 'missing oauth code' }` when I try to "Add an Organization" in the Jira Github Config. 🙃 I'll reset and try again tomorrow

**Reply by tebriel** [reply]
@Luke-Py-Walker if it's been pending for over 45 min on a small repo that's an indication that it's failed somehow in an unexpected way. The sync job is a background job and may have failed or not received the job over Bull Queue (using redis as the backing for that queue). That would be the first place I'd look.

As for the `missing oauth code` try installing your app back directly on your github organization first (from the github app's settings page), sounds like something got very confused.

**Reply by Luke-Py-Walker** [reply]
Finally got around to debugging today... probably need to update the `CONTRIBUTING.md` @tebriel 

Running `script/server` would result in a `TUNNEL_SUBDOMAIN must be set` "error" (I don't have a subdomain since the free version of `ngrok` doesn't allow it 😅 . So instead I ran `npm run dev` for my dev testing... which didn't spin up the redis worker. After that realization, I got `script/server` working by commenting out the code in `script/tunnel` that threw the missing subdomain error 😄. Another hitch occurred with `script/server` with a `User imited to 1 simultaneous ngrok client session.` error from ngrok. But I quickly realized that the `script/server` must be spinning up its own ngrok tunnel, and found the web address after coming through the printed logs. 

We should probably add a section for dealing with the missing subdomain (I'm sure that's something most people trying to contribute face). A sentence pointing out that `script/server` spins up the ngrok tunnel and a more clear log message explicitly stating the ngrok URL should also be added. 

With everything running, I'll start putting together the PR for #375. Should I put a separate PR up for updating `CONTRIBUTING.md`?

**Reply by tebriel** [reply]
Sorry that the contribution experience is so difficult there. A separate PR for the contributing markdown file would be great as it could get merged faster and wouldn't be dependent on your other PR! Thanks for putting in all this effort.

**Reply by zmoazeni** [reply]
@Luke-Py-Walker Yeah, echoing @tebriel's sentiments, please keep sharing this friction. And if you are interested, please help us out by submitting one-off PRs to help the next OSS dev that comes along. 

Thank you for persevering!
